jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/actions/DeployServerlessApplicationAction.kt

```diff
@@ -34,10 +37,14 @@ class DeployServerlessApplicationAction : DumbAwareAction(
         val templateFile = getSamTemplateFile(e) ?: throw Exception("Could not detect template file")
         val template = CloudFormationTemplate.parse(project, templateFile)
 
-        val stackDialog = DeployServerlessApplicationDialog(project, template.parameters())
+        val module = ModuleUtil.findModuleForFile(templateFile, project)
```

Comment:
Is module the right thing to tie this to? I feel like it should maybe be template file - though then we have a problem with tracking renames...

My concern with module is then we can only have one stack per module - what if they have more than one template file / stack per module?

Reply:
Moved to file level and the key is the relative path of the template file to the root of the module.

Open question: currently, we only recognize template file with the hard coded name **template.yaml** or **template.yml** for deployment. In this case, there are no way to have multiple yaml files in the same folder. Do we need to loosen the restriction for sam template files?